pkg/hostagent/pod_relatives.go

```diff
@@ -323,6 +327,8 @@ func deploymentLogger(log *logrus.Logger, dep *appsv1.Deployment) *logrus.Entry
 
 func (agent *HostAgent) deploymentAdded(obj interface{}) {
 	agent.log.Infof("deploymentAdded => ")
+	agent.indexMutex.Lock()
+	defer agent.indexMutex.Unlock()
```

Comment:
Is the lock not required for the call from L362 as well?

Reply:
Yes, missed it